projects/go-lib/src/lib/components/go-table/go-table.component.html

```diff
@@ -1,26 +1,51 @@
 <div class="go-table-container" *ngIf="showTable">
-  <table class="go-table" *ngIf="hasData(); else noDataTable">
-    <thead>
-      <tr>
-        <th *ngFor="let col of columns" class="go-table__th" [ngClass]="columnClasses(col.field)" (click)="toggleSort(col.field)">
-          <ng-container *ngIf="!col.goTableHead; else columnHeaderOutlet">{{ col.title || col.field }}</ng-container>
-          <ng-template #columnHeaderOutlet>
-            <ng-container *ngTemplateOutlet="col.goTableHead"></ng-container>
-          </ng-template>
-        </th>
-      </tr>
-    </thead>
-    <tbody>
-      <tr *ngFor="let item of localTableConfig.tableData" class="go-table__tr">
-        <td *ngFor="let col of columns" class="go-table__td" width="{{ col.width || 'auto' }}">
-          <ng-container *ngIf="!col.goTableCell; else columnCellOutlet">{{ col.getFieldData(item) }}</ng-container>
-          <ng-template #columnCellOutlet>
-            <ng-container *ngTemplateOutlet="col.goTableCell; context: { $implicit: item }"></ng-container>
-          </ng-template>
-        </td>
-      </tr>
-    </tbody>
-  </table>
+  <div class="go-table__table">
+    <table class="go-table" *ngIf="hasData(); else noDataTable">
+      <thead>
+        <tr>
+          <th *ngFor="let col of columns" class="go-table__th" [ngClass]="columnClasses(col.field)" (click)="toggleSort(col.field)">
+            <ng-container *ngIf="!col.goTableHead; else columnHeaderOutlet">{{ col.title || col.field }}</ng-container>
+            <ng-template #columnHeaderOutlet>
+              <ng-container *ngTemplateOutlet="col.goTableHead"></ng-container>
+            </ng-template>
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr *ngFor="let item of setDisplayData()" class="go-table__tr">
+          <td *ngFor="let col of columns" class="go-table__td" width="{{ col.width || 'auto' }}">
+            <ng-container *ngIf="!col.goTableCell; else columnCellOutlet">{{ col.getFieldData(item) }}</ng-container>
+            <ng-template #columnCellOutlet>
+              <ng-container *ngTemplateOutlet="col.goTableCell; context: { $implicit: item }"></ng-container>
+            </ng-template>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+  <div class="go-table-toolbar go-table-toolbar--bottom" *ngIf="showPaging()">
+    <div class="go-table-page-size">
+      <select class="go-form__select" (change)="setPerPage($event)">
+        <ng-container *ngFor="let size of localTableConfig.paging.pageSizes">
```

Comment:
What is the purpose of this `ng-container`? Couldn't this be written:
```html
<option [ngValue]="size" *ngFor="let size of localTableConfig.paging.pageSizes">{{size}}</option>
```
Additionally, I think if we use `ngValue` it prevents a potential case where the text "{{ size }}" could be evaluated as the value of this input.

Reply:
Yep, you're right here too. I've removed the `ng-container`. :+1:

Reply:
Looks like `ngValue` isn't quite right here. We could update this to use `[value]="size"` though if we wanted to avoid the above mentioned edge case.